hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractCallTransitionLogic.java

```diff
@@ -359,10 +359,10 @@ private Account extractAndValidateReceiver(
             } else {
                 if (!OK.equals(isUsableContract)) {
                     // call to non-existing contract flow
-                    validateTrue(!EVM_VERSION_0_30.equals(properties.evmVersion()), isUsableContract);
-                    validateTrue(!EVM_VERSION_0_34.equals(properties.evmVersion()), isUsableContract);
-                    validateTrue(!EVM_VERSION_0_38.equals(properties.evmVersion()), isUsableContract);
-                    validateTrue(properties.allowCallsToNonContractAccounts(), isUsableContract);
+                    validateTrue(isTokenAccount || !EVM_VERSION_0_30.equals(properties.evmVersion()), isUsableContract);
```

Comment:
do we care about direct calls to system contract addresses?

Reply:
so direct calls to 0x167 is not allowed?  I always forget whether we allow this or not.

Reply:
it not allowed, only internal calls to hedera precompiles work + token redirect calls but we allow those with this fix